File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        max_currect=max_globle=nums[0]
        for i in range(1,len(nums)):
            max_currect=max(nums[i],nums[i]+max_currect)
            if(max_currect>max_globle):
                max_globle=max_currect
        return max_globle

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxsum=nums[0];
        int sum=0;
        for(int i=0;i<nums.length;i++){
            sum=Math.max(nums[i],sum+nums[i]);
            if(maxsum<sum){
                maxsum=sum;
            }
        }
        return maxsum;
    }
}